dataflow/src/org/checkerframework/dataflow/cfg/CFGBuilder.java

```diff
@@ -4198,91 +4212,56 @@ public Node visitUnary(UnaryTree tree, Void p) {
-                        } else {
-                            assert kind == Tree.Kind.POSTFIX_DECREMENT;
-                            operNode = new NumericalSubtractionNode(operTree, exprRHS, one);
-                        }
-                        extendWithNode(operNode);
-
-                        Node narrowed = narrowAndBox(operNode, exprType);
-                        // TODO: By using the assignment as the result of the expression, we
-                        // act like a pre-increment/decrement.  Fix this by saving the initial
-                        // value of the expression in a temporary.
-                        AssignmentNode assignNode = new AssignmentNode(tree, expr, narrowed);
-                        extendWithNode(assignNode);
-                        result = assignNode;
-                        break;
-                    }
                 case PREFIX_DECREMENT:
                 case PREFIX_INCREMENT:
                     {
                         ExpressionTree exprTree = tree.getExpression();
-                        TypeMirror exprType = InternalUtils.typeOf(exprTree);
-                        TypeMirror oneType = types.getPrimitiveType(TypeKind.INT);
                         Node expr = scan(exprTree, p);
 
-                        TypeMirror promotedType = binaryPromotedType(exprType, oneType);
-
-                        LiteralTree oneTree = treeBuilder.buildLiteral(Integer.valueOf(1));
-                        handleArtificialTree(oneTree);
-
-                        Node exprRHS = binaryNumericPromotion(expr, promotedType);
-                        Node one = new IntegerLiteralNode(oneTree);
-                        one.setInSource(false);
-                        extendWithNode(one);
-                        one = binaryNumericPromotion(one, promotedType);
-
-                        BinaryTree operTree =
-                                treeBuilder.buildBinary(
-                                        promotedType,
-                                        (kind == Tree.Kind.PREFIX_INCREMENT
-                                                ? Tree.Kind.PLUS
-                                                : Tree.Kind.MINUS),
-                                        exprTree,
-                                        oneTree);
-                        handleArtificialTree(operTree);
-                        Node operNode;
-                        if (kind == Tree.Kind.PREFIX_INCREMENT) {
-                            operNode = new NumericalAdditionNode(operTree, exprRHS, one);
+                        boolean isIncrement =
+                                kind == Tree.Kind.POSTFIX_INCREMENT
+                                        || kind == Kind.PREFIX_INCREMENT;
+                        boolean isPostfix =
+                                kind == Tree.Kind.POSTFIX_INCREMENT
+                                        || kind == Kind.POSTFIX_DECREMENT;
+                        AssignmentNode compoundAssign =
```

Comment:
Also here the name `compoundAssign` is confusing, as it is an assignment node and not a compound assignment (for which there wouldn't be a Node type.)